src/audio/module_adapter/module_adapter.c

```diff
@@ -866,7 +866,7 @@ static int module_adapter_audio_stream_copy_1to1(struct comp_dev *dev)
 	return ret;
 }
 
-static int module_adapter_audio_stream_type_copy(struct comp_dev *dev)
+static inline int module_adapter_audio_stream_type_copy(struct comp_dev *dev)
```

Comment:
I'm suspicious about this. We usually assume that the compiler mostly ignores `inline` in .c files and decides by itself which functions to inline. And typically it does inline static functions that are only called once. Have you checked the resulting object? With which compiler does this make a difference?

Reply:
I wouldn't add `inline`s in these cases unless I knew that they actually improve the compiler output

Reply:
I proved inline is necessary for static function, I converted module_adapter.c obj file to asm file, then compare it with/without inline, the different is obvious, let me added somewhere to show this.

Reply:
hm, that's a pity. Which compiler was this?

Reply:
@lyakh @btian1 we still have this https://github.com/thesofproject/sof/issues/5212

Reply:
inline should only be added with:
1. only called once or very limited times.
2. timing critical calling.

for example, audio_stream_fmt_conversion this should not inline, it does not make sense, I will change it later.
it is neither timing critical calling, also called multiple times in different place.

Reply:
> @lyakh @btian1 we still have this #5212

@kv2019i ok, that explains it, thanks! That's why xcc doesn't make decisions to inline use-once static functions...

Reply:
> inline should only be added with:
> 
>     1. only called once or very limited times.
> 
>     2. timing critical calling.

3. when inlining the function hardly makes the caller larger, when the function to be inlined is very small, e.g. like with API wrappers

```
static inline int call_init(struct context *ctx)
{
  return ctx->ops.init(ctx);
}
```